dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/RateLimiterTest.java

```diff
@@ -0,0 +1,172 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import com.google.common.collect.ImmutableMap;
+import io.github.bucket4j.TimeMeter;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.atomic.AtomicInteger;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import static org.testng.Assert.*;
+
+public class RateLimiterTest {
+
+    @DataProvider(name = "rateLimitingData")
+    public static Object[][] rateLimitingData() {
+        return new Object[][]{{2, 10, 5}, {4, 16, 4}};
+    }
+
+    @DataProvider(name = "getRateLimitingData")
```

Comment:
confusing data provider name... can you also move this down to where the test is?

Reply:
changed the name.. however, still didn't bring the data and method together. IDE default formatting takes it back to the top of the class.  That'll happen with most of the devs...